### PR TITLE
feat: add mosh completion spec

### PR DESCRIPTION
**Pull request by brandonleon**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
feature, new completion spec for mosh.

**What is the current behavior? (You can also link to an open issue here)**
missing completion for mosh, mobile shell

**What is the new behavior (if this is a feature change)?**
new completion spec

**Additional info:**

### CODE REVIEW

File: src/mosh.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+      description: "Show help for mosh",
+    },
+    {
+      name: ["--client"],
+      description: "mosh client on local machine (default: \"mosh-client\")",
+    },
+    {
+      name: ["--server"],
+      description: "mosh server on remote machine (default: \"mosh-server\")",
+    },
```
Comment (mschrage) [inline src/mosh.ts:57]: Do these options take an argument?

File: src/mosh.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+      description: "do not send terminal initialization string",
+    },
+    {
+      name: ["--local"],
+      description: "run mosh-server locally without using ssh",
+    },
+    {
+      name: ["--experimental-remote-ip="],
+      description: "{local|remote|proxy}, select the method for discovering the remote IP address to use for mosh (default: \"proxy\")",
+    },
```
Comment (fedeci) [inline src/mosh.ts:125]:
```suggestion
      name: "--experimental-remote-ip",
      description: "Select the method for discovering the remote IP address to use for mosh",
      requiresSeparator: true,
      args: {
        name: "method",
        suggestions: ["local", "remote", "proxy"],
        default: "proxy",
      },
    },
```

File: src/mosh.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    {
+      name: ["--port", "-p"],
+      description: "server-side UDP port or range, (No effect on server-side SSH port)",
+    },
+    {
+      name: ["--bind-server="],
+      description: "{ssh|any|IP}, ask the server to reply from an IP address, (default: \"ssh\")",
+    },
+    {
+      name: ["--ssh="],
```
Comment (fedeci) [inline src/mosh.ts:107]:
```suggestion
      name: "--ssh",
      requiresSeparator: true,
      args: {
        name: "command",
        isCommand: true,
      },
```

File: src/mosh.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+      description: "try all network types",
+    },
+    {
+      name: ["--family=prefer-inet"],
+      description: "use all network types, but try IPv4 first [default]",
+    },
+    {
+      name: ["--family=prefer-inet6"],
+      description: "use all network types, but try IPv6 first",
+    },
```
Comment (fedeci) [inline src/mosh.ts:97]: I wonder if it's better to keep these options separate or to merge them and then duplicate `-4` and `-6`.
/cc @SeparateRecords @mschrage WDYT?

File: src/mosh.ts [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    return out
+      .split("\n")
+      .filter((line) => line.trim().startsWith("Host ") && !line.includes("*"))
+      .map((host) => ({
+        name: host.split(" ")[1],
+        description: "SSH host",
+        priority: 90,
+      }));
+  },
+};
```
Comment (fedeci) [inline src/mosh.ts:35]: You can avoid to duplicate these generators and import them from the ssh spec:
```suggestion
import { knownHosts, configHosts } from "./ssh"
```

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks for this spec 🚀! I left a couple comments. Can you also run `npm run lint:fix` before pushing so that we can get the CI not to fail?

File: src/mosh.ts (hunk 2 above)
Comment (brandonleon) [reply src/mosh.ts:125]: Agreed

File: src/mosh.ts (hunk 5 above)
Comment (brandonleon) [reply src/mosh.ts:35]: I missed this feature. I like this method much better.

File: src/mosh.ts (hunk 4 above)
Comment (brandonleon) [reply src/mosh.ts:63]: I considered that as well. I decided to stick to how the --help had it laid out, but that's not to say collapsing it all into -4 and -6 isn't a better option. 

I'm actually leaning toward your suggestion of making it simpler.

Comment (mschrage) [reply src/mosh.ts:63]: I think these options should be split out. Any option that has a `=` character in the name is really is the combination of _an option_ and _an argument_.

Comment (fedeci) [reply src/mosh.ts:63]: Let's merge them in a single option + argument then!

Comment (brandonleon) [reply src/mosh.ts]: I'll get that done this evening.

File: src/mosh.ts [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+      name: ["--predict=always", "-a"],
+      description: "Use local echo even on fast links",
+    },
+    {
+      name: ["--predict=never", "-n"],
+      description: "Never use local echo",
+    },
+    {
+      name: "--predict=experimental",
+      description: "Aggressively echo even when incorrect",
```
Comment (fedeci) [inline src/mosh.ts:38]: Could you apply the same changes of `--family` to `--predict` too?

File: src/mosh.ts [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+      name: "-4",
+      description: "Use IPv4 only",
+    },
+    {
+      name: "-6",
+      description: "Use IPv6 only",
+    },
+    {
+      name: "--family",
+      description: "Network Type",
```
Comment (fedeci) [inline src/mosh.ts:50]: If an equals is required between the option and the argument you should add this option:
```suggestion
      description: "Network Type",
      requiresSeparator: true,
```

File: src/mosh.ts [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+      args: {
+        name: "auto",
+        description: "Autodetect network type for single-family hosts only",
```
Comment (fedeci) [inline src/mosh.ts:53]:
```suggestion
        name: "type",
```

**Review by fedeci — CHANGES_REQUESTED** [review]
Great, a couple more changes and we are done!

**Comment by brandonleon** [comment]
Will fix the last few errors tomorrow.

**Review by fedeci — APPROVED** [review]
Thanks!

**Pull request merged**